### PR TITLE
Add support for HKHeartbeatSeriesSample

### LINKED ISSUE #121: Add support for HKHeartbeatSeriesSample
HR series samples are useful for accessing the underlying IBI measurements used to compute the heart rate variability.  This is crucial in health research settings where a researcher may want to analyze the values used to compute the hrv or use their own method instead of apple's SDNN approach.

I can tackle this after finishing the clinical records PR unless someone else wants to work on it sooner.

### PR DESCRIPTION
**Pull request by klandell**
## Description

Adds support for reading the HKHeartbeatSeriesSample type.  I think HealthKit allows writing this type as well, but I don't have the bandwidth to build that our right now.

Fixes #121

## Type of change

- [x] New feature (non-breaking change which adds functionality)
- [x] This change requires a documentation update

## Checklist:

- [x] My code follows the style guidelines of this project
- [x] I have performed a self-review of my own code
- [x] I have made corresponding changes to the documentation
- [x] I have checked my code and corrected any misspellings

### CODE REVIEW

**Comment by alfonsosn** [comment]
Nit: you should probably squash your commits into a single one. It will make your PR easier to review.

**Comment by klandell** [comment]
@alfonsosn Done

File: RCTAppleHealthKit/RCTAppleHealthKit+Methods_Vitals.m [hunk 1]
```diff
@@ -151,6 +151,108 @@ - (void)vitals_getHeartRateVariabilitySamples:(NSDictionary *)input callback:(RC
...
+                     @"endDate" : endDateString,
+                     @"heartbeatSeries": @[]
+                 };
+                NSMutableDictionary *mutableElem = [elem mutableCopy];
+                [data addObject:mutableElem];
+                
+                // create an array to hold the series data which will be fetched asynchronously from healthkit
+                NSMutableArray *seriesData = [NSMutableArray arrayWithCapacity:sample.count];
+                
+                // mow define the data handler for the HeartbeatSeriesQuery
```
Comment (macelai) [inline RCTAppleHealthKit/RCTAppleHealthKit+Methods_Vitals.m:216]: Typo

**Review by macelai — CHANGES_REQUESTED** [review]
Great job as usual

File: RCTAppleHealthKit/RCTAppleHealthKit+Methods_Vitals.m (hunk 1 above)
Comment (klandell) [reply RCTAppleHealthKit/RCTAppleHealthKit+Methods_Vitals.m]: Oops. Typo fixed.

**Pull request merged**